### PR TITLE
Warn on imposing max rows and fix for curve

### PR DESCRIPTION
**Pull request by ahuang11**
Closes https://github.com/holoviz/hvplot/issues/1406

Adds a warning and checks if kind == curve to prevent random sampling; only sample head.

### CODE REVIEW

File: hvplot/ui.py [hunk 1]
```diff
@@ -688,10 +688,22 @@ def _plot(self):
         if len(df) > MAX_ROWS and not (
             self.kind in KINDS['stats'] or kwargs.get('rasterize') or kwargs.get('datashade')
         ):
-            df = df.sample(n=MAX_ROWS)
+            if self.kind == 'line':
+                param.main.param.warning(
```
Comment (maximlt) [inline hvplot/ui.py:692]: How about displaying these warnings in the alert box in addition/in place of as a programmatic warning?

**Review by maximlt — COMMENTED** [review]
- I feel like `MAX_ROWS` could be increased, 10000 is kind of low specially considering that now HoloViews uses the webGL backend by default for Bokeh. How about 100000?
- Just for discussion for now, do you think we should expose this kind of setting in the explorer? Maybe in an advanced tab?

File: hvplot/ui.py (hunk 1 above)
Comment (hagaishalevaei) [reply hvplot/ui.py:692]: I suggest to do df = df.sample(n=MAX_ROWS).sort_index()`. Otherwise the line plot will not work as it should.

Comment (ahuang11) [reply hvplot/ui.py:692]: I disagree with sort_index; the x may not always be index.

Comment (hagaishalevaei) [reply hvplot/ui.py:692]: good point. is there an option to add sort_index as an option? 
something like `df.hvplot.explorer(x='x', y='y', kind='line', sort_index=True)` 
where the defualt is `sort_index=False` ?

Comment (ahuang11) [reply hvplot/ui.py:692]: I think it probably makes more sense to do `sort_values("x")` when sampling for line.

File: hvplot/tests/testui.py [hunk 2]
```diff
@@ -395,3 +395,22 @@ def test_explorer_geo_revise_kind(kind_tuple):
...
+    N = 10001
+    x = np.linspace(0.0, 6.4, num=N)
+    y = np.sin(x) + 10
+    df = pd.DataFrame({'x': x, 'y': y})
+    ui = hvplot.explorer(df, x='x', y='y', by=['#'], kind='line')
+    assert ui._data.equals(df.head(MAX_ROWS))
+
+
+def test_max_rows_sample():
+    N = 10001
```
Comment (ahuang11) [inline hvplot/tests/testui.py:410]:
```suggestion
    N = 100001
    x = np.linspace(0.0, 6.4, num=N)
    y = np.sin(x) + 10
    df = pd.DataFrame({'x': x, 'y': y})
    ui = hvplot.explorer(df, x='x', y='y', by=['#'], kind='line')
    assert ui._data.equals(df.head(MAX_ROWS))


def test_max_rows_sample():
    N = 100001
```

**Review by maximlt — APPROVED** [review]

**Pull request merged**